src/dbus/server/dbus_thread_object_rcp.cpp

```diff
@@ -2009,21 +2009,17 @@ void DBusThreadObjectRcp::DeactivateEphemeralKeyModeHandler(DBusRequest &aReques
 {
     otError error        = OT_ERROR_NONE;
     auto    threadHelper = mHost.GetThreadHelper();
+    bool    retain_active_session;
+    auto    args = std::tie(retain_active_session);
 
     VerifyOrExit(mBorderAgent.GetEphemeralKeyEnabled(), error = OT_ERROR_NOT_CAPABLE);
 
-    switch (otBorderAgentGetState(threadHelper->GetInstance()))
+    SuccessOrExit(DBusMessageToTuple(*aRequest.GetMessage(), args), error = OT_ERROR_INVALID_ARGS);
+    if (!retain_active_session)
     {
-    case OT_BORDER_AGENT_STATE_STOPPED:
-        error = OT_ERROR_FAILED;
-        break;
-    case OT_BORDER_AGENT_STATE_ACTIVE:
-        error = OT_ERROR_INVALID_STATE;
-        break;
-    case OT_BORDER_AGENT_STATE_STARTED:
-        otBorderAgentClearEphemeralKey(threadHelper->GetInstance());
-        break;
+        otBorderAgentDisconnect(threadHelper->GetInstance());
     }
+    otBorderAgentClearEphemeralKey(threadHelper->GetInstance());
```

Comment:
To better match the doc in xml, may update as below?
 ```
if (!retain_active_session && otBorderAgentGetState(threadHelper->GetInstance()) >= OT_BORDER_AGENT_STATE_ACTIVE)
 {
    // Ephemeral Key will be cleared after the session is disconnected
     otBorderAgentDisconnect(threadHelper->GetInstance());
 }
else
{
    otBorderAgentClearEphemeralKey(threadHelper->GetInstance());
}
```

Reply:
otBorderAgentClearEphemeralKey also needs to be called after disconnect(when BA is start state), if there is no secure session, calling otBorderAgentDisconnect wouldn't clear the ephemeral key.

Reply:
it's covered in the` else` already - when no secure session, otBorderAgentClearEphemeralKey will be called

Reply:
I think it's ok to let otBorderAgentDisconnect take care of the different BA states (e.g. no-op when BA state is not `OT_BORDER_AGENT_STATE_ACTIVE`), the current code looks good and simple to me.

Reply:
As offline discussion, fine to keep it as is, and may update API reference for otBorderAgentClearEphemeralKey and otBorderAgentDisconnect for forcely deactivation operation